actors/util/adt/set.go

```diff
@@ -0,0 +1,74 @@
+package adt
+
+import (
+	"io"
+
+	cid "github.com/ipfs/go-cid"
+	hamt "github.com/ipfs/go-hamt-ipld"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+var setValue hamtSetValue
+
+func init() {
+	setValue = hamtSetValue(1)
+}
+
+type hamtSetValue uint64
+
+func (s hamtSetValue) MarshalCBOR(w io.Writer) error {
+	if _, err := w.Write(cbg.CborEncodeMajorType(cbg.MajUnsignedInt, uint64(s))); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
It took me a while to figure out what's going on here, but I get it now - a sentinel value because you can't use nil.
- I think you can use `struct{}` (regardless of how we CBOR it), which will be cheaper in Go and more obviously empty, just like when used as a map value
- since it always encodes to the same byte, hardcode it rather than compute it
- is there prior art for using the value 1? We could use any single-byte encoding, such as the same thing as `EmptyReturn`, again a better representation of "nothing". 

Basically I think you can move/rename `EmptyReturn` to `adt.EmptyValue` and use it here.

Reply:
Prior art here: https://github.com/filecoin-project/lotus/blob/master/chain/actors/actor_storagepower.go#L751

> Basically I think you can move/rename EmptyReturn to adt.EmptyValue and use it here.

will give it a try, good idea!